test/unit/concepts/comparison/three_way_comparable.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Christopher Di Bella.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+#include "cjdb/concepts/comparison/three_way_comparable.hpp"
+
+#include <version>
+#if defined(__cpp_lib_three_way_comparison) and __cpp_lib_three_way_comparison >= 201711
+
+#include "cjdb/concepts/core/common_reference_with.hpp"
+#include "cjdb/concepts/core/convertible_to.hpp"
+#include "three_way_comparable.hpp"
+
+template<typename T, typename Cat, bool expected>
+constexpr void check_three_way_comparable_impl() noexcept
+{
+	static_assert(cjdb::three_way_comparable<T, Cat> == expected);
+}
+
+template<typename T, cjdb::common_reference_with<T> U, typename Cat, bool expected>
```

Comment:
I feel like you just want to do `typename U`, then you can check for cases where `T` and `U` don't have a common reference and that that is `false`.

Reply:
This particular test should only permit things with a common reference through, so I think it should remain. The case where there isn't a common reference was missing, and has now been added.